Rewrite abstract (#69)
greenelab/text_mined_hetnet_manuscript: content/01.abstract.md

## Abstract {.page_break_before}

### Background

Knowledge graphs support biomedical research efforts by providing contextual information for biomedical entities, constructing networks, and supporting the interpretation of high-throughput analyses. 
These databases are populated via some form of manual curation, which is challenging to scale with an exponentially rising publication rate. 
Data programming is a paradigm that circumvents this arduous manual process by combining databases with simple rules and heuristics written as label functions, which are programs designed to annotate textual data automatically. 
Unfortunately, writing a useful label function requires substantial error analysis and is a nontrivial task that takes multiple days per function.
This bottleneck makes populating a knowledge graph with multiple nodes and edge types practically infeasible.
We sought to accelerate the label function creation process by evaluating how label functions can be re-used across multiple edge types.

> We obtained entity-tagged abstracts from Pubtator Central and subsetted these entities to only contain compounds, genes, and diseases mentions.

Reviewer: ```suggestion
We obtained entity-tagged abstracts and subsetted these entities to only contain compounds, genes, and disease mentions.
```

> Using an existing knowledge graph, Hetionet v1, we extracted sentences that mentioned the following biomedical relationships: Compound treats Disease (CtD), Gene interacts Gene (GiG), Compound binds Gene (CbG) and Disease associates Gene (DaG).

Reviewer: ```suggestion
We extracted sentences that mentioned certain biomedical entities that were contained in a previously described knowledge graph.
```

> We trained a baseline model that used database-only label functions and then used a sampling approach to measure how well adding edge-specific or edge-mismatch label function combinations improve over our baseline.

Reviewer: These two sentences feel like results. Can you move this + maybe the two sentences above that are a bit concrete about what you did to results and we can take another look?

Unfortunately, writing a useful label function requires substantial error analysis and is a nontrivial task that takes multiple days per function.
This bottleneck makes populating a knowledge graph with multiple nodes and edge types practically infeasible.
Thus, we sought to accelerate the label function creation process by evaluating how label functions can be re-used across multiple edge types.

### Results

We obtained entity-tagged abstracts and subsetted these entities to only contain compounds, genes, and disease mentions.
We extracted sentences containing co-mentions of certain biomedical entities contained in a previously described knowledge graph, Hetionet v1.
We trained a baseline model that used database-only label functions and then used a sampling approach to measure how well adding edge-specific or edge-mismatch label function combinations improved over our baseline.
Next, we trained a discriminator model to detect sentences that indicated a biomedical relationship and then estimated the number of edge types that could be recalled and added to Hetionet v1.

> We found that adding additional edge-specific rather than edge-mismatch label functions often improves text annotation.

Reviewer: ```suggestion
We found that adding edge-mismatch label functions rarely improved relationship extraction, while control edge-specific label functions did.
```

> We found that adding additional edge-specific rather than edge-mismatch label functions often improves text annotation.

Reviewer: ```suggestion
We found that adding edge-mismatch label functions rarely improved relationship extraction, while control edge-specific label functions did.
```

> We found that adding additional edge-specific rather than edge-mismatch label functions often improves text annotation.

Reviewer: ```suggestion
We found that adding edge-mismatch label functions rarely improved relationship extraction, while control edge-specific label functions did.
```

> An exception to this trend was two edge types, Compound-binds-Gene and Gene-interacts-Gene, as these edges showed signs of transferability.

Reviewer: ```suggestion
There were two exceptions to this trend, Compound-binds-Gene and Gene-interacts-Gene, which both indicated physical relationships and showed signs of transferability.
```

> We found the discriminative model performance strongly depends on generated annotations.

Reviewer: ```suggestion
Across the scenarios tested, discriminative model performance strongly depends on generated annotations.
```

Using the best discriminative model for each edge type, we recalled close to 30% of established edges within Hetionet v1.

### Conclusions

> Our results show that this framework can incorporate novel edges into our source knowledge graph.

Reviewer: ```suggestion
Our results show that this framework can incorporate novel edges into our source knowledge graph.
However, results with label function transfer were mixed.
Only label functions describing very similar edge types supported improved performance when transferred.
```

> We expect that the continued development of this strategy has the potential to swiftly populate biomedical knowledge graphs with new discoveries, ensuring that these resources include cutting-edge results.

Reviewer: ```suggestion
We expect that the continued development of this strategy may provide essential building blocks to populating biomedical knowledge graphs with discoveries, ensuring that these resources include cutting-edge results.
```
